app/controllers/projects_controller.rb

```diff
@@ -543,77 +543,206 @@ def order_investigations
 
   def publish_to_csh
```

Comment:
i think the most of code of the "publish_to_csh" should be move to a module in `lib/nfdi4health`,  the code for preparing and parsing json could be reused by other resource to publish to CSH.

Reply:
https://github.com/nfdi4health/ldh/blob/52132af3321b9e76bed66eeeab1f07c428da8ad2/app/controllers/projects_controller.rb#L552

https://github.com/nfdi4health/ldh/blob/52132af3321b9e76bed66eeeab1f07c428da8ad2/app/controllers/investigations_controller.rb#L156

https://github.com/nfdi4health/ldh/blob/52132af3321b9e76bed66eeeab1f07c428da8ad2/app/controllers/studies_controller.rb#L364